Deepgram/Abstractions/AbstractRestClient.cs

```diff
@@ -36,15 +43,74 @@ internal AbstractRestClient(string apiKey = "", DeepgramClientOptions? options =
     /// <typeparam name="T">Type of class of response expected</typeparam>
     /// <param name="uriSegment">request uri Endpoint</param>
     /// <returns>Instance of T</returns>
-    public virtual async Task<T> GetAsync<T>(string uriSegment, CancellationToken cancellationToken = default, Dictionary<string, string>? addons = null)
+    public virtual async Task<T> GetAsync<T>(string uriSegment, CancellationTokenSource? cancellationToken = null,
+    Dictionary<string, string>? addons = null, Dictionary<string, string>? headers = null)
+    {
+        try
+        {
+            // if not defined, use default timeout
+            if (cancellationToken == null)
+            {
+                cancellationToken = new CancellationTokenSource();
+                cancellationToken.CancelAfter(Constants.DefaultRESTTimeout);
+            }
+
+            // create request message and add custom query parameters
+            NoopSchema? parameter = null;
+            var request = new HttpRequestMessage(HttpMethod.Get, QueryParameterUtil.FormatURL(uriSegment, parameter, addons));
+
+            // add custom headers
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    request.Headers.Add(header.Key, header.Value);
+                }
+            }
+
+            // do the request
+            var response = await _httpClient.SendAsync(request, cancellationToken.Token);
+            response.EnsureSuccessStatusCode();
+            var result = await HttpRequestUtil.DeserializeAsync<T>(response);
+
+            return result;
+        }
+        catch (Exception ex)
+        {
+            Log.Exception(_logger, "GET", ex);
+            throw;
+        }
+    }

```

Comment:
Acknowledge, but this allows me to use a timeout per REST call that is also thread-safe. Using a standard `CancellationToken` only allows me to create a single global timeout for all REST calls. This call can only reliably be set once because setting it multiple times might yield unexpected results.